lightning/src/ln/channelmanager.rs

```diff
@@ -4101,6 +4101,32 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		}
 	}
 
+	/// Called to respond to a request to open a channel after the
+	/// [`events::Event::OpenChannelRequest`] has been triggered.
+	///
+	/// If `accept_channel` is set to true, the requested channel will be opened. If
+	/// `accept_channel` is false, the open channel request will be rejected.
+	///
+	/// The `temporary_channel_id` parameter indicates which channel the reponse is for.
+	///
+	/// [`events::Event::OpenChannelRequest`]: crate::util::events::Event::OpenChannelRequest
+	pub fn respond_to_open_channel_request(&self, accept_channel: bool, temporary_channel_id: [u8; 32]) -> Result<(), APIError> {
+		let _persistence_guard = PersistenceNotifierGuard::notify_on_drop(&self.total_consistency_lock, &self.persistence_notifier);
+
+		if let Some(counterparty_node_id) = self.internal_get_counterparty_node_id(temporary_channel_id) {
```

Comment:
This is fundamentally race-y - you take the lock in `internal_get_counterparty_node_id`, check if the channel is there, and then release the lock, then go into `accept_requested_channel` and take the lock again. Anyway, this will be easy to remove with the above suggestion.

Reply:
Thanks, I'll fix it through the above suggestion.